packages/opencensus-core/src/resource/resource.ts

```diff
@@ -0,0 +1,192 @@
+/**
+ * Copyright 2018, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {StringUtils} from '../internal/string-utils';
+
+/**
+ * Resource represents a resource, which capture identifying information about
+ * the entities for which signals (stats or traces) are reported. It further
+ * provides a framework for detection of resource information from the
+ * environment and progressive population as signals propagate from the core
+ * instrumentation library to a backend's exporter.
+ */
+export class Resource {
+  /**
+   * The type identifier for the resource.
+   */
+  private readonly type: string|null;
+  /**
+   * A map of labels that describe the resource.
+   */
+  private readonly labels: StringMap;
+
+  private static readonly MAX_LENGTH = 255;
+  private static readonly LABEL_LIST_SPLITTER = ',';
```

Comment:
Using value-naming for constant could help with using it in whole file, and separator is more readable (COMA_SEPARATOR), the same for LABEL_KEY_VALUE_SPLITTER